public/static/js/version.js

```diff
@@ -161,6 +161,9 @@ mciModule.controller('VersionController', function($scope, $rootScope, $location
           if (cell.task_end_details.type == 'system') {
             return 'system-failed';
           }
+          if (cell.task_end_details.type == 'setup') {
+            return 'setup-failed';
+          }
```

Comment:
It looks like these lines are duplicated several times. Is it an option to move these lines into some service?

Reply:
Yea, there is quite a bit of duplication here. Parts of this are for determining css classes, and parts are strings that are part of the UI. Given that there is a plan in works to refactor the UI, I'm not too eager to go into cleaning this up.